src/swap/saga.ts

```diff
@@ -0,0 +1,114 @@
+
+    // Check that our guaranteedPrice is within 2%, maxSwapSlippagePercentage, of of the price
+    const maxSlippagePercent: number = yield select(maxSwapSlippagePercentageSelector)
+    const priceDiff: number = yield call(
+      getPercentageDifference,
+      +action.payload.unvalidatedSwapTransaction.price,
+      +action.payload.unvalidatedSwapTransaction.guaranteedPrice
+    )
+    if (priceDiff >= maxSlippagePercent) {
+      yield put(swapPriceChange())
+      return
+    }
+
+    // Set contract kit, wallet address and normalizer
+    const kit: ContractKit = yield call(getContractKit)
+    const walletAddress: string = yield call(getConnectedUnlockedAccount)
+    const normalizer = new TxParamsNormalizer(kit.connection)
+
+    // Approve transaction
+    yield put(swapApprove())
+    Logger.debug(TAG, `Starting to swap approval for address: ${walletAddress}`)
+    const rawApproveTx = { ...action.payload.approveTransaction }
+    applyChainIdWorkaround(rawApproveTx, yield call([kit.connection, 'chainId']))
+    const approveTx: CeloTx = yield call(normalizer.populate.bind(normalizer), rawApproveTx)
+    const approveTxo = buildTxo(kit, approveTx)
+    yield call(
+      sendTransaction,
+      approveTxo,
+      walletAddress,
+      newTransactionContext(TAG, 'Swap/Approve')
+    )
+
+    // Query the execute swap endpoint
+    const amountType =
+      action.payload.userInput.updatedField === Field.TO ? 'buyAmount' : 'sellAmount'
+    const amount = action.payload.unvalidatedSwapTransaction[amountType]
+    const params = {
+      buyToken: action.payload.unvalidatedSwapTransaction.buyTokenAddress,
+      sellToken: action.payload.unvalidatedSwapTransaction.sellTokenAddress,
+      [amountType]: amount,
+      userAddress: walletAddress,
+    }
+    const queryParams = new URLSearchParams({ ...params }).toString()
+    const requestUrl = `${networkConfig.executeSwapUrl}?${queryParams}`
+    const response: Response = yield call(fetchWithTimeout, requestUrl)
+    if (!response.ok) {
+      Logger.error(TAG, `Swap failed with status: ${response.status}`)
+      yield put(swapError())
+      return
+    }
+    const responseJson: { validatedSwapTransaction: SwapTransaction } = yield call([
+      response,
+      'json',
+    ])
+
+    // Execute transaction
+    yield put(swapExecute())
+    Logger.debug(TAG, `Starting to swap execute for address: ${walletAddress}`)
+    const rawExecuteTx = responseJson.validatedSwapTransaction
+    applyChainIdWorkaround(rawExecuteTx, yield call([kit.connection, 'chainId']))
+    const executeTx: CeloTx = yield call(normalizer.populate.bind(normalizer), rawExecuteTx)
+    const executeTxo = buildTxo(kit, executeTx)
+    yield call(
+      sendTransaction,
+      executeTxo,
+      walletAddress,
+      newTransactionContext(TAG, 'Swap/Execute')
+    )
+    yield put(swapSuccess())
+  } catch (error) {
+    Logger.error(TAG, 'Error while swapping', error)
```

Comment:
would also suggest `Logger.debug` here

Reply:
Changed in 60fead26fafbb504e9d50c5f906545c71086b9d7!